pkg/adapter/mtping/runner.go

```diff
@@ -191,11 +192,23 @@ func makeEvent(source *sourcesv1.PingSource) (cloudevents.Event, error) {
 func (a *cronJobsRunner) newPingSourceClient(source *sourcesv1.PingSource) (adapter.Client, error) {
 	var env adapter.EnvConfig
 	if a.clientConfig.Env != nil {
-		env = adapter.EnvConfig{
-			Namespace:      source.GetNamespace(),
-			Name:           a.clientConfig.Env.GetName(),
-			EnvSinkTimeout: fmt.Sprintf("%d", a.clientConfig.Env.GetSinktimeout()),
+		if source.Status.Auth != nil {
+			env = adapter.EnvConfig{
+				Namespace:              source.GetNamespace(),
+				Name:                   a.clientConfig.Env.GetName(),
+				EnvSinkTimeout:         fmt.Sprintf("%d", a.clientConfig.Env.GetSinktimeout()),
+				Audience:               source.Status.SinkAudience,
+				OIDCServiceAccountName: source.Status.Auth.ServiceAccountName,
+			}
+		} else {
+			env = adapter.EnvConfig{
+				Namespace:      source.GetNamespace(),
+				Name:           a.clientConfig.Env.GetName(),
+				EnvSinkTimeout: fmt.Sprintf("%d", a.clientConfig.Env.GetSinktimeout()),
+				Audience:       source.Status.SinkAudience,
+			}
 		}
```

Comment:
```suggestion
		env = adapter.EnvConfig{
			Namespace:              source.GetNamespace(),
			Name:                   a.clientConfig.Env.GetName(),
			EnvSinkTimeout:         fmt.Sprintf("%d", a.clientConfig.Env.GetSinktimeout()),
			Audience:               source.Status.SinkAudience,
		}
		
		if source.Status.Auth != nil {
			env.OIDCServiceAccountName = source.Status.Auth.ServiceAccountName
		}
```
nit. So we don't need to duplicate the code.